amps/ags/rm-community/rm-community-repo/source/java/org/alfresco/module/org_alfresco_module_rm/action/RecordsManagementActionServiceImpl.java

```diff
@@ -285,6 +301,17 @@ public RecordsManagementActionResult executeRecordsManagementAction(NodeRef node
             throw new AlfrescoRuntimeException(msg);
         }
 
+        if (dispositionService.isFrozenOrHasFrozenChildren(nodeRef) && retentionActions.contains(name.toLowerCase())) {
+            String msg = I18NUtil.getMessage(MSG_NODE_FROZEN, rmAction.getName());
+
+            if (logger.isDebugEnabled())
```

Comment:
I think I mentioned in another PR that it would be good to start using slf4j to avoid these extra checks.